werkzeug/routing.py

```diff
@@ -765,7 +765,9 @@ def match(self, path):
                 # tells the map to redirect to the same url but with a
                 # trailing slash
                 if self.strict_slashes and not self.is_leaf and \
-                   not groups.pop('__suffix__'):
+                        not groups.pop('__suffix__') and \
+                        (method is None or self.methods is None or
```

Comment:
No, that is not the same for the `methods` one. In your case it will fail if `methods` is `None`. Which is not what we want. If methods is empty it should fail, but if methods is None it should not (right?). 
The simplification to the truth value of method seems fine though. It could cause issues when supplying `''`, but that is not a valid method anyway.